action/protocol/poll/slasher.go

```diff
@@ -548,3 +594,26 @@ func filterCandidates(
 	}
 	return verifiedCandidates, nil
 }
+
+// currentEpochProductivity returns the map of the number of blocks produced per delegate of current epoch
+func currentEpochProductivity(sr protocol.StateReader, start uint64, end uint64) (map[string]uint64, error) {
+	stats := make(map[string]uint64)
+	epochMeta, err := epochMetaFromDB(sr)
+	if err != nil {
+		return nil, err
+	}
+	for _, blockmeta := range epochMeta.BlockMetas {
+		if blockmeta.Height < start {
+			return nil, errors.New("block meta height is out of range")
```

Comment:
how could this happen?

Reply:
it shouldn't happen, just in case I wrote it